### PR TITLE
chore(EMS-3501-3497): iterations - application population tech improvements

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:
This PR updates the `getPopulatedApplication` API function so that each application relationship and related fetching/mapping/population has it's own function.

## Resolution :heavy_check_mark:
- Update various existing API test helper functions to accept an optional `data` param.
- Create new API test helper functions to get and create:
  - `broker`
  - `business`
  - `buyerRelationship`
  - `buyer`
  - `country`
  - `declaration`
  - `eligibility`
  - `exportContract`
  - `policyContact`
  - `privateMarket`
- Create new API helper functions to:
  - `getBrokerById`
  - `getBusinessById`
  - `getBuyerById`
  - `getBuyerRelationshipById`
  - `getBuyerTradingHistoryById`
  - `getCompanyAddressById`
  - `getCompanyById`
  - `getCompanyDifferentTradingAddress`
  - `getCompanySicCodesByCompanyId`
  - `getCountryById`
  - `getCoverPeriodById`
  - `getDeclarationById`
  - `getEligibilityById`
  - `getExportContractById`
  - `getPolicyById`
  - `getPrivateMarketById`
  - `getSectionReviewById`
  - `getTotalContractValueById`
- Create new population helper functions:
  - `getPopulatedBuyer`
  - `getPopulatedCompany`
  - `getPopulatedEligibility`
  - `getPopulatedExportContract`
- Update `getPopulatedApplication` to consume the new helper functions.
- 

## Miscellaneous :heavy_plus_sign:
- Add some missing API application types.
- Minor documentation improvements.
- Create a DRY `mockInvalidId` for all API tests.

### CODE REVIEW

File: src/api/helpers/get-buyer-trading-history-by-id/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    createdRelationship = await buyerTradingHistory.create(context) as ApplicationBuyerTradingHistory;
+  });
+
+  it('should return a buyer trading history by ID', async () => {
+    const result = (await getBuyerTradingHistoryById(context, createdRelationship.id));
+
+    expect(result.id).toEqual(createdRelationship.id);
+  });
+
+  describe('when a buyer is not found', () => {
```
Comment (Zainzzkk) [inline src/api/helpers/get-buyer-trading-history-by-id/index.test.ts:24]: buyer trading history

File: src/api/helpers/get-policy-by-id/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  beforeAll(async () => {
+    context = getKeystoneContext();
+  });
+
+  beforeEach(async () => {
+    createdPolicy = await policy.create({ context }) as ApplicationPolicy;
+  });
+
+  it('should return a total contract value by ID', async () => {
```
Comment (Zainzzkk) [inline src/api/helpers/get-policy-by-id/index.test.ts:18]: policy

File: src/api/helpers/get-policy-contact-by-id/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  beforeAll(async () => {
+    context = getKeystoneContext();
+  });
+
+  beforeEach(async () => {
+    createdPolicyContact = await policyContact.create(context) as ApplicationPolicyContact;
+  });
+
+  it('should return a total contract value by ID', async () => {
```
Comment (Zainzzkk) [inline src/api/helpers/get-policy-contact-by-id/index.test.ts:18]: policy contact

File: src/api/helpers/get-populated-export-contract/get-populated-agent/index.test.ts [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    const result = await getPopulatedAgent(context, createdAgent.id);
+
+    expect(result.id).toEqual(createdAgent.id);
+    expect(result.service.id).toEqual(createdService.id);
+    expect(result.service.charge.id).toEqual(createdServiceCharge.id);
+  });
+
+  describe('when an agent is not found', () => {
+    it('should throw an error', async () => {
+      const invalidId = 'invalid-id';
```
Comment (Zainzzkk) [inline src/api/helpers/get-populated-export-contract/get-populated-agent/index.test.ts:53]: const at the top? and in other files too as repeated

File: src/api/test-helpers/declaration.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    return err;
+  }
+};
+
+/**
+ * Get declaration test helper
+ * Get an declaration by ID
+ * @param {Context} context: KeystoneJS context API
+ * @param {String} declarationId: Declaration ID
+ * @returns {Promise<ApplicationExportContract>} Declaration
```
Comment (Zainzzkk) [inline src/api/test-helpers/declaration.ts:28]: wrong return type

File: src/api/helpers/get-populated-export-contract/get-populated-agent/index.test.ts (hunk 4 above)
Comment (ttbarnes) [reply src/api/helpers/get-populated-export-contract/get-populated-agent/index.test.ts:53]: Updated all instances of "invalid ID" stubs in the API

**Review by Zainzzkk — APPROVED** [review]

**Pull request merged**